external/java/src/main/java/ch/epfl/dedis/lib/crypto/bn256/GFpPool.java

```diff
@@ -0,0 +1,72 @@
+package ch.epfl.dedis.lib.crypto.bn256;
+
+import java.util.LinkedList;
+
+/**
+ * This singleton class keeps a pool of different GFp. Instead of creating manually new GFp objects that will then
+ * be eventually deleted by the GC, one can use this class which will keep no longer used objects and re-use them.
+ * If the pool is empty a new object is created. We initialize GFp instances to ZERO.
+ */
+public class GFpPool {
```

Comment:
does the pool need to be public? i think the classes in the same package can access it if it's just `class GFpPool`